File: Product of array except self/product.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        prefix_product = 1 
        postfix_product = 1 
        result = [0 for i in range(n)]
        for i in range (n):
            result[i]=prefix_product
            prefix_product *= nums[i]
        for j in range(n-1,-1,-1):
            result[j] *= postfix_product 
            postfix_product *= nums[j]
        return result 

File: Product of array except self/product.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> res(n, 1);
        int left_product = 1;
        for (int i = 0; i < n; ++i) {
            res[i] = left_product;
            left_product *= nums[i];
        }
        int right_product = 1;
        for (int i = n - 1; i >= 0; --i) {
            res[i] *= right_product;
            right_product *= nums[i];
        }

        return res;
    }
};
